### ISSUE TITLE
http://meteor.local makes crypto webapi unusable!!

### ISSUE THREAD

**Issue by dotnetwise** [meteor/meteor#5671]
`crypto webapi` are only valid over `HTTPS` or localhost
since meteor only works over dummy `http://meteor.local` it will never be able to execute any `crypto apis`

```js
function createAndSaveAKeyPair() {
        // Returns a promise.
        // Takes no input, yields no output to then handler.
        // Side effect: updates keyPair in enclosing scope with new value.

        return window.crypto.subtle.generateKey(
            {
                name: "RSA-OAEP",
                modulusLength: 2048,
                publicExponent: new Uint8Array([1, 0, 1]),  // 24 bit representation of 65537
                hash: {name: "SHA-256"}
            },
            true,   // can extract it later if we want
            ["encrypt", "decrypt"]).
        then(function (key) {
            keyPair = key;
            return key;
        });
    }
createAndSaveAKeyPair();
```

causes this eception
```js
Uncaught (in promise) DOMException: Only secure origins are allowed. http://goo.gl/lq4gCo
```

I am testing this suing the `meteor add crosswalk` package

**Comment by martijnwalraven** [comment]
I hadn't encountered this before, but it seems there is no way around it as long as we're using `http://meteor.local`. 

Fortunately, I'm currently rewriting the Meteor file serving and hot code push plugin, and part of this work is switching to an embedded web server. That means we will use `http://localhost` instead, which according to [the page you linked to](http://www.chromium.org/Home/chromium-security/prefer-secure-origins-for-powerful-new-features) should be considered a secure origin.

**Comment by dotnetwise** [comment]
That would be amazing!

Is this part of the core? Any ETA when it will be released? `meteor 1.3`, earlier?
Can we apply it as an atmosphere package from the dev branch earlier?

**Comment by martijnwalraven** [comment]
It will be part of core in Meteor 1.3. It is too tightly interwoven to be released as an atmosphere package unfortunately. But it shouldn't be too long until Meteor 1.3.

**Comment by dotnetwise** [comment]
Perhaps worth tagging this and other into a `meteor 1.3` milestone so we'd know what other goodies will be included?

**Comment by Exilz** [comment]
@martijnwalraven wonderful news, would this allow us to use wkwebview too ?

**Comment by AlbinoGeek** [comment]
To put a little back news on this issue:

- Chrome decided that all places non-HTTPS are not secure origins and disabled all dangerous APIs.

This means you cannot use the following:

- Key pair generation (or key management)
- Webcam or UserMedia control
- Audio and Video sources
- Remote Capture
- Cross-capture (MIDI or other control devices)
- User Filesystem (incl. file dropping after download, or upload from)
- Geolocation and other user-local devices such as Gyroscopes.
- Fullscreen

The page you want is actually [this one}(https://www.chromium.org/Home/chromium-security/deprecating-powerful-features-on-insecure-origins) which explains which features were deprecated in the insecure space, and why they were.

**Comment by dotnetwise** [comment]
so how far is this to be released?

**Comment by martijnwalraven** [comment]
This has been fixed in Meteor 1.3 beta (see https://github.com/meteor/meteor/issues/6050).
